lib/wickedbase.pm

```diff
@@ -234,6 +234,31 @@ sub get_test_result {
     }
 }
 
+sub get_wicked_logs {
+    my ($self, $prefix) = @_;
+    my $dir_name = $self->{name} . '_' . $prefix;
+    my $logs_dir = "/tmp/$dir_name";
+    record_info('Logs', "Collecting logs in $logs_dir");
+    script_run("mkdir -p $logs_dir");
+    script_run("date +'%Y-%m-%d %T.%6N' > $logs_dir/date");
+    script_run("journalctl -b -o short-precise|tail -n +2|grep wicked > $logs_dir/journalctl.log");
+    script_run("wicked ifstatus --verbose all > $logs_dir/wicked_ifstatus.log 2>&1");
+    script_run("wicked show-config > $logs_dir/wicked_config.log 2>&1");
+    script_run("wicked show-xml > $logs_dir/wicked_xml.log 2>&1");
+    script_run("ip addr show > $logs_dir/ip_addr.log 2>&1");
+    script_run("ip route show table all > $logs_dir/ip_route.log 2>&1");
+    script_run("tar -C /tmp/ -cvzf $dir_name.tar.gz $dir_name");
+    eval {
+        upload_logs("$dir_name.tar.gz", failok => 0, log_name => " ");
+        1;
+    } or do {
```

Comment:
Never have seen this style but it reads nicely :)

Reply:
I am flexible, can be changed if you have a proposal :)

Reply:
I just meant to say I like it, nothing to change